### ISSUE TITLE
Standard shortcuts for copy/cut/paste don't work in MacOSX

### ISSUE THREAD

**Issue by pierpaolocira** [akasolace/HO#177]
Hi,
when I went to authorise CHPP, I copied the authorisation code from the website and tried to paste it into the related textfield of HO.
Anyway that field doesn't allow to paste neither via the keybord shortcut nor by opening the contextual menu (right click).
I had to manually type the code.

I use 1.437 on Mac OSX Sierra.

**Comment by akasolace** [comment]
Hey @pierpaolocira,  is ctr+c / ctr+v working ? (I am using ctrl and not apple special key as it should normally be the case)

**Comment by pierpaolocira** [comment]
Hi @akasolace, in which folder should the HO db saved? I would like to (temporarily) move it, in order to launch the application and have gain the OAuth screen (so that I can place it again in its place after testing).

Just to mention, some tests I did:
- I was not really able to open the application from source files, because of a lot of Exceptions linked to the way Swing is mapped to the native MacOSX components: it more or less starts, but then every event linked to a listener trigger the "Error" Screen.
I just forced the application to directly open the OAuth screen, and I was successfully able to paste text via Cmd-V.
- If I open the application (from the binaries from MacOS that I can doenload here) and go to a screen containing a textfiled (e.g. the "Proxy" configuration screen) I cannot paste via Cmd-V... but I can if I use Ctrl-V shortcut
I have the feeling that this is a more broad issue on MacOSX, relating all the screens and the copy/cut/paste shortcuts... might it be the way it is packaged as a binary... or maybe something like this: https://stackoverflow.com/questions/7252749/how-to-use-command-c-command-v-shortcut-in-mac-to-copy-paste-text

What do you think?

**Comment by akasolace** [comment]
@pierpaolocira,   
> in which folder should the HO db saved? 
 in MacOS I believe the db folder is located in the user home directory

> What do you think?
Regarding the errors you got I don't know as I don't recall similar issue when I tested it in a mac virtual machine.
If I understand correctly what you are saying, if app launched from the IDE or java command,  cmd+v is working but if the app is launched directly via the binary then it CTR+C which is working and not CMD-V, correct ?

If it is the case it is more annoying if it behaves differenty once packaged. I would have hoped it is simply an issue wiht the listener  (as per the SO link you put)

**Comment by pierpaolocira** [comment]
@akasolace let me do some additional tests, and I'll give you a more detailed and clear picture.

I'm sorry not to be able to reply more quickly, but I'm still struggling with the aforementioned issue Java+Mac+IDE (it seems that there is a bug in the Swing implementation under Aqua, but I was not able to solve neither by updating java, not by upgrading it and launching in compatibility "java 8" mode).
Can I just ask you how HO is packaged? Does it embed a specific JRE, or it uses the system JRE?

Thanks.

**Comment by akasolace** [comment]
@pierpaolocira  HO is using the system JRE.  Which IDE are you using?  Last time I tried, it was in a VM using Intellij and it was working fine...

@mistermax80  maybe you can help here.

**Comment by mistermax80** [comment]
@pierpaolocira Could you try to set you IDE (on MacOs) as the following screenshot:
![Schermata_2018-12-15_alle_10 02 41](https://user-images.githubusercontent.com/15519907/61892352-e69c1300-af0b-11e9-8fb0-146c0f1b20e4.png)

**Comment by pierpaolocira** [comment]
@mistermax80 thank you very much. Indeed adding the VM Options I was able to run it properly from the IDE. I'm not a big expert in Awt/Swing and I didn't find something like that in google (but now I see that it is also reported in the Gradle build).

@akasolace I can now tell you that I was wrong in one of my previous replies, sorry.

**There is no difference in running HO via the IDE or the via the binary ufficially packaged by you: CTRL has to be used instead of CMD in order to copy/cut/paste on MacOSX**
So it doesn't regard only the OAuthDialog, but also other textfield (for example I double checked in SkillTester (field Age) or in Proxy Configuration).
If possible I will change the title of the issue.

**Comment by pierpaolocira** [comment]
@akasolace 
I read again the StackOverflow reply that I pasted, and I think it's the cause (i.e. it seems that we are using a different L&F).

Indeed `UIManager.getLookAndFeel().getName()` returns `Nimbus`.
But default value in MacOSX should be `Mac OS X`, that is the value I get if I set it at the beginning of the `main` in `Ho.java`.

Furthermore... If I, for example, write the following code beside JTextField, it start working:
```
InputMap im = jtfAge.getInputMap(WHEN_FOCUSED);
im.put(KeyStroke.getKeyStroke(KeyEvent.VK_V, KeyEvent.META_DOWN_MASK), DefaultEditorKit.pasteAction);
```
And... If I force this L&F not to work, HO sets another one (namely Metal) that successfully enables those shortcuts.

I think I can work on a patch, in order to force again those shortcuts in case a L&F is changed under MacOS...

**Comment by pierpaolocira** [comment]
It seemed to be quite easy... I hope it complies with project guidelines.
Please remember to test it under a system different that MacOSX (I cannot do that) in order to be sure that no regressions are introduced.

**Comment by akasolace** [comment]
Thank you I will test and merge tomorrow ... 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by akasolace** [comment]
closed by #180
